src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonPropertyInfoOfT.cs

```diff
@@ -408,13 +411,25 @@ internal override bool ReadJsonAsObject(ref ReadStack state, ref Utf8JsonReader
             return success;
         }
 
-        internal override void SetExtensionDictionaryAsObject(object obj, object? extensionDict)
+        internal sealed override void SetExtensionDictionaryAsObject(object obj, object? extensionDict)
```

Comment:
I believe this applies to all overrides since `JsonPropertyInfo<T>` is sealed on the type level.

Reply:
This predates your changes, but couldn't this method be replaced with calling the untyped `Set` delegate?